climada/trajectories/impact_calc_strat.py

```diff
@@ -0,0 +1,291 @@
+from climada.engine.impact_calc import ImpactCalc
+from climada.entity.exposures.base import Exposures
+from climada.entity.impact_funcs.impact_func_set import ImpactFuncSet
+from climada.hazard.base import Hazard
+
+__all__ = ["ImpactCalcComputation"]
+
+
+class ImpactComputationStrategy(ABC):
+    """
+    Interface for impact computation strategies.
+
+    This abstract class defines the contract for all concrete strategies
+    responsible for calculating and optionally modifying with a risk transfer,
+    the impact computation, based on a set of inputs (exposure, hazard, vulnerability).
+
+    It revolves around a `compute_impacts()` method that takes as arguments
+    the three dimensions of risk (exposure, hazard, vulnerability) and return an
+    Impact object.
+    """
+
+    @abstractmethod
+    def compute_impacts(
+        self,
+        exp: Exposures,
+        haz: Hazard,
+        vul: ImpactFuncSet,
+        risk_transf_attach: Optional[float] = None,
+        risk_transf_cover: Optional[float] = None,
+        calc_residual: bool = True,
+    ) -> Impact:
+        """
+        Calculates the total impact, including optional risk transfer application.
+
+        Parameters
+        ----------
+        exp : Exposures
+            The exposure data.
+        haz : Hazard
+            The hazard data (e.g., event intensity).
+        vul : ImpactFuncSet
+            The set of vulnerability functions.
+        risk_transf_attach : float, optional
+            The attachment point (deductible) for the global risk transfer mechanism.
+            If None, default to 0.
+        risk_transf_cover : float, optional
+            The cover limit for the risk transfer mechanism. If None, the cover
+            is assumed to be infinite (only the attachment applies).
+        calc_residual : bool, default=True
+            If True, the function returns the residual impact (after risk transfer).
+            If False, it returns the transferred impact (the part covered by the
+            risk transfer).
+
+        Returns
+        -------
+        Impact
+            An object containing the computed total impact matrix and metrics.
+
+        See Also
+        --------
+        ImpactCalcComputation : The default implementation of this interface.
+        """
+        ...
+
+
+class ImpactCalcComputation(ImpactComputationStrategy):
+    r"""
+    Default impact computation strategy using the core engine of climada.
+
+    This strategy first calculates the raw impact using the standard
+    :class:`ImpactCalc` logic and then applies a global risk transfer mechanism.
+
+    This risk transfer is distinct and non-exclusive with the risk transfer
+    defined through the Exposures object. While in the Exposures case, the transfer
+    is defined on a per coordinate basis, this one is applied total impact on the whole region
+    considered, and proportionally rescales the impacts per coordinate accordingly.
```

Comment:
```suggestion
    is defined on a per coordinate basis, this one is applied total impact on the whole region
    considered, and proportionally rescales the impacts per coordinate accordingly.
```
What does the `rescales the impacts per coordinate accordingly` mean ? Attachment and cover would just apply for the whole exposure right? What is being rescaled?

Reply:
The original code from Victor changed the value of the impact matrix by proportionally rescaling them such that their sum corresponded to the transfered (residual) risk after this global risk transfer was applied.

Reply:
I do not understand this. I think we should consider what makes sense, not necessarily what was implemented tentatively in the past. 
Can we find at least some evidence that it is a good idea to equally distribute global risk transfer money?

Reply:
I agree with you about the little sense it makes.
But actually, maybe we just don't want that? How do you want us to consider this global transfer?
I mean, either we change the impact matrix or we don't. And rescaling the values is what seems to be the most straightforward if we do change it.

We can also not change the matrix and save the information elsewhere, but then it means we have to completely rethink the way we compute risk metrics.

I personally would go with just removing this feature (at least for now, until it is clearer).

Reply:
Good idea. I would keep the code either in it but not accessible, or in a separate branch.